sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -111,6 +111,14 @@ statement
         (AS? query)?                                                   #createHiveTable
     | CREATE TABLE (IF NOT EXISTS)? target=tableIdentifier
         LIKE source=tableIdentifier locationSpec?                      #createTableLike
+    | replaceTableHeader ('(' colTypeList ')')? tableProvider
+        ((OPTIONS options=tablePropertyList) |
+        (PARTITIONED BY partitioning=transformList) |
+        bucketSpec |
```

Comment:
Should bucketing be added using `BUCKET BY`? Or should we rely on `bucket` as a transform in the `PARTITIONED BY` clause?

Reply:
Same as https://github.com/apache/spark/pull/24798#discussion_r291395947 - this is copied from the create table spec.